go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -353,11 +390,44 @@ func (oc *SecondaryLocalnetNetworkController) newRetryFramework(
 }
 
 func (oc *SecondaryLocalnetNetworkController) localnetPortNetworkNameOptions() map[string]string {
-	localnetLSPOptions := map[string]string{
-		"network_name": oc.GetNetworkName(),
+	return map[string]string{
+		"network_name": oc.getNetworkName(),
+	}
+}
+
+func (oc *SecondaryLocalnetNetworkController) checkBridgeMapping(obj interface{}) error {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
+	}
+
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	openvSwitch, err := ovsops.GetOpenvSwitch(oc.ovsClient)
+	if err != nil {
+		return err
+	}
+	ovnBridgeMappings := openvSwitch.ExternalIDs["ovn-bridge-mappings"]
+
+	networkName := oc.getNetworkName()
+	bridgeMappings := strings.Split(ovnBridgeMappings, ",")
+	for _, bridgeMapping := range bridgeMappings {
+		m := strings.Split(bridgeMapping, ":")
+		if m[0] == networkName {
+			return nil
+		}
 	}
```

Comment:
maybe encapsulate this in an helper func ?

Reply:
no need imho, since we are already in a small helper